container/lxd/network.go

```diff
@@ -501,19 +498,16 @@ func DevicesFromInterfaceInfo(interfaces corenetwork.InterfaceInfos, machineID s
 // TODO (manadart 2018-06-21) We want to support nictype=macvlan too.
 // This will involve interrogating the parent device, via the server if it is
 // LXD managed, or via the container.NetworkConfig.DeviceType that this is
-// being generated from. If the hostIfaceName argument is not empty, LXD will
-// use its value as the name of the container's virtual eth device on the host.
-// Otherwise, the host interface will be assigned a random unique value by LXD.
-func newNICDevice(deviceName, parentDevice, hostIfaceName, hwAddr string, mtu int) device {
+// being generated from.
+// It's important to note, that the host interface will be assigned a random
```

Comment:
I would move this information up to `networkDevicesFromConfig`, omitting that "it is important to note"...